osquery/tables/networking/windows/interface_details.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <map>
+#include <sstream>
+#include <string>
+
+#include <stdlib.h>
+
+#include <boost/algorithm/string/join.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/tables/system/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+void genInterfaces(QueryData& results_data) {
+  std::stringstream ss;
+  ss << "SELECT * FROM Win32_NetworkAdapter";
+
+  WmiRequest request(ss.str());
+  if (request.getStatus().ok()) {
+    std::vector<WmiResultItem>& results = request.results();
+    for (const auto& result : results) {
+      Row r;
+      Status s;
+      long lPlaceHolder;
+      bool bPlaceHolder;
+      std::string sPlaceHolder;
+      std::vector<std::string> vPlaceHolder;
+      unsigned __int64 ulPlaceHolder;
+
+      s = result.GetString("AdapterType", sPlaceHolder);
+      r["adapter_type"] = SQL_TEXT(sPlaceHolder);
```

Comment:
For most of these:
```cpp
result.GetString("AdapterType", r["adapter_type"]);
```

and call it a day. I'm not sure if `SQL_TEXT` is needed, if it is maybe we can change/overload the WMI functions later on and come back to refactor these calls.